includes/MediaWiki/Request/EditRequestParser.php

```diff
@@ -39,11 +38,9 @@ public function __construct(
 		$this->minimalItemInput = $minimalItemInput;
 	}
 
-	public function parseRequestInterface( RequestInterface $request ): EditRequest {
-		$reconcile = json_decode(
-			$request->getPostParams()['reconcile'],
-			true
-		);
+	public function parseRequestBody( array $requestBody ): EditRequest {
+		$reconcile = $requestBody['reconcile'];
+
 		if ( !is_array( $reconcile ) ) {
 			throw new LocalizedHttpException(
 				MessageValue::new( 'wikibasereconcileedit-editendpoint-invalid-reconcile-json' ),
```

Comment:
This should probably be a different error message, since the JSON syntax is now checked by `JsonBodyValidator`.

Reply:
done